app/controllers/articles_controller.rb

```diff
@@ -195,6 +196,24 @@ def admin_unpublish
     end
   end
 
+  def discussion_lock_confirm
```

Comment:
> Since the articles controller is "too far gone" at this point 

The [broken windows theory of software development](https://medium.com/@matryer/broken-windows-theory-why-code-quality-and-simplistic-design-are-non-negotiable-e37f8ce23dab) strikes again.

Not criticizing your decision here, I think your argument is valid and reasonable. But it also serves as a warning as to why we want to ensure new code doesn't get into a state like this.